packages/components/src/Select.tsx

```diff
@@ -0,0 +1,240 @@
+import _ from 'lodash/fp';
+import React from 'react';
+import RSelect, {
+  components as RSelectComponents,
+  GroupBase as RSelectGroupBase,
+  MultiValue as RSelectMultiValue,
+  Props as RSelectProps,
+  PropsValue as RSelectPropsValue,
+  SingleValue as RSelectSingleValue,
+} from 'react-select';
+
+import { useUniqueId } from './hooks/useUniqueId';
+import { icon } from './icon';
+import { useThemeFromContext } from './theme';
+
+let colors: {
```

Comment:
Pretty sure there is a better way to handle this (this is declared in L226)

Reply:
We'd definitely want to get rid of this global variable. It looks like this could lead to theme colors leaking across different theme providers. Practically, it might not be an issue since apps likely have only one theme provider, but still.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4681/commits/41236dc750d730a2ad3587fec20d10c03c4dc2cb.